### PR TITLE
Add a noop() function

### PR DESCRIPTION
**Pull request by phanan**
This PR adds a `noop()` function which might be useful in certain cases.

### CODE REVIEW

File: src/Functional/Noop.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+namespace Functional;
+
+/**
+ * A no-operation function.
+ */
+function noop()
```
Comment (lstrojny) [inline src/Functional/Noop.php]: Would the return type be `void` or `null`?

Comment (phanan) [reply src/Functional/Noop.php]: Shouldn't it be `void` as it does nothing?

Comment (lstrojny) [reply src/Functional/Noop.php]: Strictly speaking it would then be illegal to use it in a map

Comment (phanan) [reply src/Functional/Noop.php]: There will always be an illegal case for any return type, right? 🤔

**Comment by lstrojny** [comment]
One thing I am wondering: should this be a higher order function so that it returns a function?

**Comment by phanan** [comment]
What would be the benefit of returning a function over a simple `void`?

**Comment by lstrojny** [comment]
Something like `map($v, noop())` but it wouldn’t be in line with `id()`

**Pull request merged**